packages/@atjson/renderer-commonmark/src/index.ts

```diff
@@ -283,10 +288,18 @@ export default class CommonmarkRenderer extends Renderer {
     if (text.length === 0) {
       return before + after;
     } else {
+      // When there is no surrounding whitespace,
+      // we will allow for an incorrect nesting of
+      // bold / italic to maintain the general intent
+      // over correctness.
+      let hasInnerMarkup =
+        context.children.length === 1 && before === "" && after === "";
+
       if (
         !context.previous &&
         !context.next &&
-        context.parent instanceof Italic
+        context.parent instanceof Italic &&
+        !hasInnerMarkup
       ) {
         return `${before}__${text}__${after}`;
```

Comment:
In this case it is probably safer to use `**` as a delimiter

Reply:
Actually, we need to keep `__` in this case 😭